music21/humdrum/spineParser.py

```diff
@@ -2199,197 +2222,186 @@ def hdStringToNote(contents):
     >>> humdrum.spineParser.flavors['JRP'] = storedFlavors  #_DOCS_HIDE
 
     '''
-
-    # http://www.lib.virginia.edu/artsandmedia/dmmc/Music/Humdrum/kern_hlp.html#kern
+    dur_regex, dotCount, matchedNote, step, accountedFor = _noteMemos(contents)
+
+    if dur_regex:
+        foundNumber, foundRational = dur_regex.groups()
```

Comment:
all of this takes an already very complex set of routines and makes them even more complex -- now with a set of nested if statements.  Together with the rewriting of the count routines, I can't make heads or tails of what has changed.  Please refactor into a routine that gets the duration from contents and one that gets the pitch from contents.  I have no idea what "accountedFor" means or what a dur_regex is supposed to represent.